src/v2/recommender.py

```diff
@@ -142,7 +142,7 @@ def extract_version(data):
             return version
 
         def has_valid_version(data):
-            return extract_version(data) != ""
+            return extract_version(data) not in ["", "-1"]
```

Comment:
Invalid version will have "-1" or -1 (string or integer), can we add -1 integer also in list to be on safer side..

Reply:
no harm is adding -1 as well. LGTM

Reply:
Too much of defensive programming :) Converted the version to string before testing for invalid version.